tests/liteclient_test.cc

```diff
@@ -200,15 +182,45 @@ class TufRepoMock {
   }
 
  private:
+  const std::string root_;
   ImageRepo repo_;
-  std::string port_;
-  std::string url_;
-  boost::process::child process_;
   Uptane::Target latest_;
 };
 
-std::string TufRepoMock::ServerPath;
+class DeviceGatewayMock {
+ public:
+  static std::string RootPath;
+ public:
+  DeviceGatewayMock(const OSTreeMock& ostree_mock,
+                    const TufRepoMock& tuf_repo_mock):
+                    ostree_mock_{ostree_mock},
+                    tuf_repo_mock_{tuf_repo_mock},
+                    port_{TestUtils::getFreePort()},
+                    url_{"http://localhost:" + port_},
+                    process_{RootPath, "--port", port_, "--ostree", ostree_mock_.path(), "--tuf-repo", tuf_repo_mock_.path()} {
+    TestUtils::waitForServer(url_ + "/");
+    LOG_INFO << "Device Gateway is running on port " << port_;
+  }
+  ~DeviceGatewayMock() {
+    process_.terminate();
```

Comment:
maybe check process_.running() before calling terminate and wait?

Reply:
the required checks are inside of these methods.